### PR TITLE
Reduce usages of SearchContext#mapperService

### PR DESCRIPTION
**Pull request by javanna**
We recently removed getMapperService from QueryShardContext in the attempt to prevent consumers from depending on the whole MapperService. SearchContext still has that problem although it is easier to solved as it can delegate to QueryShardContext for the most part, which is what this commit does for most of the existing usages.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/query/QueryPhase.java [hunk 1]
```diff
@@ -429,8 +429,7 @@ private static Query tryRewriteLongSort(SearchContext searchContext, IndexReader
         // check if this is a field of type Long or Date, that is indexed and has doc values
         String fieldName = sortField.getField();
         if (fieldName == null) return null; // happens when _score or _doc is the 1st sort field
-        if (searchContext.mapperService() == null) return null; // mapperService can be null in tests
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/search/query/QueryPhase.java:432]: this made me nervous , but I tried removing, I do expect test failures caused by it, but let's see. Same for other changes in the test classes.

**Comment by nik9000** [comment]
Oh boy! #64249 shares some of this. I'll review this PR this afternoon or tomorrow and compare.

File: server/src/main/java/org/elasticsearch/search/query/QueryPhase.java (hunk 1 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/search/query/QueryPhase.java:432]: :+1:

**Review by nik9000 — APPROVED** [review]
Actually, yeah. I just reviewed and it looks perfect. Presuming none of the sneaky tests catch anything :+1:

**Pull request merged**